Lib/re.py

```diff
@@ -281,7 +288,10 @@ def _compile(pattern, flags):
     p = sre_compile.compile(pattern, flags)
     if not (flags & DEBUG):
         if len(_cache) >= _MAXCACHE:
-            _cache.clear()
+            try:
+                _cache.popitem(False)
```

Comment:
Sorry that I'm late to the game in reviewing this, and thanks for fixing this.  One comment, should this be

`_cache.popitem(last=False)`

for readability?  (I think if history had been different, we would have made that a keyword-only argument.)

Reply:
Maybe. This wouldn't harm performance, because sre_compile.compile() few lines above is much more expensive. Open a new PR for changing this if you will.

Reply:
I'm reworking my deferred compilation branch to add a new function, so I might just fold this simple change into that branch.

Reply:
Never mind; I'm abandoning the deferred compilation branch, so  #3791